### PR TITLE
python310Packages.ale-py: init at 0.8.0

### PR DESCRIPTION
**Pull request by BillHuang2001**
###### Description of changes

ale-py (Arcade-Learning-Environment) is a python package that allows researchers and hobbyists to develop AI agents for Atari 2600 games.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ale-py/default.nix [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+, pybind11
+, pytestCheckHook
+, python
+, pythonOlder
+, setuptools
+, typing-extensions
+, wheel
+, zlib
+}:
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ale-py/default.nix:22]:
```suggestion
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/ale-py/default.nix [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  pname = "ale-py";
+  version = "0.8.0";
+  format = "pyproject";
+  rev-short = "d59d006";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ale-py/default.nix:26]:
```suggestion
```

File: pkgs/development/python-modules/ale-py/default.nix [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    SDL2
+    setuptools
+    wheel
+    pybind11
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    gcc
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ale-py/default.nix:46]:
```suggestion
  nativeBuildInputs = [
    cmake
    gcc
  ];

  buildInputs = [
    zlib
    SDL2
    setuptools
    wheel
    pybind11
  ];
```

Isn't pybind11, wheel and setuptools run on the build machine? Then they all go into nativeBuildInputs.

Also does this always require gcc? Then we should use gccStdenv.

File: pkgs/development/python-modules/ale-py/default.nix [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  checkInputs = [
+    pytestCheckHook
+    gym
+  ];
+
+  patches = [
+    # don't download pybind, use local pybind
+    ./cmake-pybind11.patch
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ale-py/default.nix:64]:
```suggestion
```

File: pkgs/development/python-modules/ale-py/default.nix [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    substituteInPlace pyproject.toml \
+      --replace 'dynamic = ["version"]' 'version = "${version}"'
+    substituteInPlace setup.py \
+      --replace 'subprocess.check_output(["git", "rev-parse", "--short", "HEAD"], cwd=here)' 'b"${rev-short}"'
+    cat setup.py
+  '';
+
+  buildPhase = ''
+    mkdir -p dist
+    ${python.interpreter} -m pip wheel --verbose --no-index --no-deps --no-clean --no-build-isolation --wheel-dir dist ../
```
[the remark this answers is not in the document]
Comment (BillHuang2001) [reply pkgs/development/python-modules/ale-py/default.nix]: Yes. Now `dontUseCmakeConfigure=true`.

[the remark this answers is not in the document]
Comment (BillHuang2001) [reply pkgs/development/python-modules/ale-py/default.nix]: Updated, cmake setup hook is disable, and normal buildPhase is used.

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.ale-py

File: pkgs/development/python-modules/ale-py/default.nix [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  dontUseCmakeConfigure = true;
+
+  pythonImportsCheck = [ "ale_py" ];
+
+  meta = with lib; {
+    description = "a simple framework that allows researchers and hobbyists to develop AI agents for Atari 2600 games";
+    homepage = "https://github.com/mgbellemare/Arcade-Learning-Environment";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ billhuang ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ale-py/default.nix:79]:
```suggestion
    maintainers = with maintainers; [ billhuang ];
    broken = stdenv.isDarwin; # fails to link with missing library
```

File: pkgs/development/python-modules/ale-py/default.nix [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+, importlib-metadata
+, importlib-resources
+, lib
+, ninja
+, numpy
+, pybind11
+, pytestCheckHook
+, python
+, pythonOlder
+, setuptools
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ale-py/default.nix:16]:
```suggestion
, setuptools
, stdenv
```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.ale-py

**Pull request merged**